src/vehicle.cpp

```diff
@@ -1008,8 +1008,8 @@ void vehicle::smash( map &m, float hp_percent_loss_min, float hp_percent_loss_ma
         int roll = dice( 1, 1000 );
         int pct_af = ( percent_of_parts_to_affect * 1000.0f );
         if( roll < pct_af ) {
-            float dist =  damage_size == 0.0f ? 1.0f :
-                          clamp( 1.0f - trig_dist( damage_origin, part.precalc[0] ) / damage_size, 0.0f, 1.0f );
+            double dist = damage_size == 0.0 ? 1.0 :
+                          std::clamp( 1.0 - trig_dist( damage_origin, part.precalc[0] ) / damage_size, 0.0, 1.0 );
```

Comment:
I keep forgetting we have to wait on c++17 for this
```suggestion
                          clamp( 1.0 - trig_dist( damage_origin, part.precalc[0] ) / damage_size, 0.0, 1.0 );
```